cocos/core/splash-screen.ts

```diff
@@ -173,13 +173,17 @@ export class SplashScreen {
             device.capabilities.clipSpaceSignY, swapchain.surfaceTransform);
         const dw = swapchain.width; const dh = swapchain.height;
         const refW = dw < dh ? dw : dh;
+        const deviceAspect = dw / dh;
+        const designRes = window._CCSettings.designResolution;
```

Comment:
It not working now, please see `settings.querySettings`

Reply:
ok, I ended up adding the design resolution to the settings using the `settings.querySettings` call